README.md

```diff
@@ -70,14 +59,213 @@ the `extension.json` file looks like so, and can include one or more HLAs.
+
+Note - your class can either process analog data or digital data, but not both. A class may handle as many metrics as you want though. If you want to implement both digital and analog measurements, you will need at a minimum two classes.
+
+The constructor will be called with an array of requested measurements, which are taken from the extension.json file. In this example, that will be `["averageVoltage", "maxVoltage"]`.
+
+Your constructor needs to pass this array to the base class:
+
+```py
+def __init__(self, requested_measurements):
+    super().__init__(requested_measurements)
+```
+
+Immediately after construction, the method `def process_data(self, data):` will be called one or more times.
+
+The Saleae Logic software stores collected data in chunks. To keep python processing performant, the Logic software passes these blocks, or sections of these blocks, one at a time to your measurement. If the requested measurement range does not line up with the internal block storage, the objects passed to python will already be adjusted to the measurement range, so no work needs to be done to handle this condition.
+
+This makes it impossible to know exactly how much data will be needed for the given measurement range the first time `process_data` is called. Be sure to update the internal state of your class in such a way that this isn't a problem. For example, when computing the average analog value over a range, it would be best to hold the sum of all values passed to `process_data` and the total count of samples in data members, and only compute the average in the `measure` function.
+
+`def process_data(self, data):`. This function takes a parameter `data` which differs between analog and digital measurements.
+
+For analog measurements, `data` is an instance of the Saleae class `AnalogData`, which is an iterable class with the properties `sample_count` and `samples`.
+`sample_count` is a number, and is the number of analog samples in the data instance. Note - this might not be the total number of analog samples passed to your measurement, since `process_data` may be called more than once if the user selected range spans multiple chunks.
+
+`samples` is a [numpy](https://numpy.org/) [ndarray](https://docs.scipy.org/doc/numpy/reference/arrays.ndarray.html). For information handling this type, please refer to the numpy documentation.
+
+The `process_data` function should not return a value. Instead, it should update the internal state of your class, such that the `measure` function can produce your measurement's results.
+
+For digital measurement classes, the `data` parameter is an instance of the iterable Saleae class `DigitalData`. Each iteration returns a pair of values - the current time relative to the start of the capture, as a floating point number of seconds, and the current bit state as a boolean. (true = signal high).
```

Comment:
We return the time as a `GraphTime` object, which currently only allows subtracting from another `GraphTime` (which produces a floating point number of seconds).

Reply:
good catch, I didn't look closely enough at the implementation. Updated!